src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -24,18 +28,27 @@
     private HttpServletResponse response;
     @Mock
     private RequestDispatcher requestDispatcher;
+    @Mock
+    private ArrayListProductDao productDao;
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
 
+    @InjectMocks
     private ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
     public void setup(){
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+        when(productDao.findProducts()).thenReturn(Arrays.asList(product1, product2));
     }
 
     @Test
     public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
 
+        verify(productDao).findProducts();
```

Comment:
Verify that this list is attached to the request object, not just is being retrieved.